app/src/models/hash.ts

```diff
@@ -0,0 +1,5 @@
+type HashableType = number | string | boolean | undefined | null
+
+export function createHash(items: HashableType[]) {
```

Comment:
I was wondering if there is an easy way of specifying a type (that is not a class) to enforce that Hashes are created with this function.

Something like `type Hash = string` doesn't work because you can still create a `string` however you want and use it instead. Does TypeScript have anything to help with this? 🤔

Reply:
No, not that I'm aware of. In TypeScript types are considered equal if they're semantically equivalent which leads to hacks like https://github.com/desktop/desktop/blob/d008892180353fd8e4417cc5c3c8478b9a6aafa3/app/src/lib/api.ts#L112-L122 in order to prevent an `IAPIRepository` from being passed as an `IAPIFullRepository`.

This is particularly easy to get tripped up on when dealing with types containing only optional fields since they're essentially assignable  by any object which shares at least one compatible field signature.

Reply:
Oh, wow, that's a good one!! 🤯 Very interesting, thanks!